react/components/PickupDetailsModal.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react'
+import { TranslateEstimate } from 'vtex.shipping-estimate-translator'
+import { FormattedPrice } from 'vtex.formatted-price'
+import { PickupOption, BusinessHour } from 'vtex.checkout-graphql'
+import { Modal, Divider } from 'vtex.styleguide'
+import { FormattedMessage, defineMessages, FormattedTime } from 'react-intl'
+
+const messages = defineMessages({
+  details: {
+    defaultMessage: 'Details',
+    id: 'store/checkout.shipping.pickupPointsModal.details',
+  },
+  weekDay0: {
+    defaultMessage: 'Sunday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay0',
+  },
+  weekDay1: {
+    defaultMessage: 'Monday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay1',
+  },
+  weekDay2: {
+    defaultMessage: 'Tuesday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay2',
+  },
+  weekDay3: {
+    defaultMessage: 'Wednesday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay3',
+  },
+  weekDay4: {
+    defaultMessage: 'Thursday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay4',
+  },
+  weekDay5: {
+    defaultMessage: 'Friday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay5',
+  },
+  weekDay6: {
+    defaultMessage: 'Saturday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay6',
+  },
+  weekDay1to5: {
+    defaultMessage: 'Monday to Friday',
+    id: 'store/checkout.shipping.pickupPointsModal.weekDay1to5',
+  },
+})
+
+const formatHour = (hour: string) => {
+  const hoursNumbers = hour.split(':').map((t) => parseInt(t, 10))
+  // eslint-disable-next-line prefer-spread
+  const time = new Date().setHours.apply(new Date(), hoursNumbers)
+
+  return <FormattedTime value={time} hour="numeric" />
+}
+
+interface Props {
+  pickupOption: PickupOption
+  setShowPickupModal: (showPickupModal: boolean) => void
+  showPickupModal: boolean
+}
+
+const PickupDetailsModal: React.VFC<Props> = ({
+  pickupOption,
+  showPickupModal,
+  setShowPickupModal,
+}) => {
+  return (
+    <Modal
+      isOpen={showPickupModal}
+      onClose={() => setShowPickupModal(false)}
+      title={<FormattedMessage {...messages.details} />}
+    >
+      <div className="lh-copy">
+        <Divider orientation="horizontal" />
+        <div className="flex-column mt4 mb4">
+          <div className="flex justify-between mb0">
+            <div className="fw6 f3">{pickupOption.friendlyName}</div>
```

Comment:
use semantic elements where it makes sense to

```suggestion
            <p className="fw6 f3 mv0">{pickupOption.friendlyName}</p>
```